### ISSUE TITLE
Transition from GI to Accession

### ISSUE THREAD

**Issue by ekrell** [cbirdlab/charybdis#46]
NCBI's GI numbers have been deprecated for a long time now, but were still supported while developing Charybdis.
Many steps and scripts in the pipeline rely on GIs.

Currently, GIs are still working with the NCBI tools. For example, we can still get lists of mitochondrial and environmental sequences to filter the blast databased with "-gilist". There is an option for using an accession list instead, so the transition should be easy. 

Where the pipeline fails is `OTUvsTubes`. There are two problems with using GIs. 
- GIs are used to query taxonomic IDs (TAXIDs), using NCBI taxonomy database. But now it requires accessions instead.
- We do this lookup using the `CHNOSZ` package, that I believe has not been updated to support accessions. 

What we should do eventually (will take quite a while):
- Fully update the pipeline to support accessions. Remove any reference to GIs
- Query TAXIDs using the `taxonomizr` package. 

What we should do right now (quick and dirty): 
- Inside `OTUvsTubes`, convert any GI to an accession
- Probably using: ftp://ftp.ncbi.nlm.nih.gov/genbank/livelists/gi2acc_mapping/ 
- Query TAXIDs using the `taxonomizr` package.

**Comment by ekrell** [comment]
# Generate mapping file between OTU's GIs and Accessions

### Install gi2acc

```bash
# Create gi2acc directory (assuming you are at Charybdis root)
mkdir -p data/gi2acc/
cd data/gi2acc
# Conversion python script
wget ftp://ftp.ncbi.nlm.nih.gov/genbank/livelists/gi2acc_mapping/gi2accession.py
# Python dependency
sudo pip install lmdb
# Conversion database
wget ftp://ftp.ncbi.nlm.nih.gov/genbank/livelists/gi2acc_mapping/gi2acc_lmdb.db.gz
gunzip gi2acc_lmdb.db.gz
# Test it out (ctrl-z to exit)
python gi2accession.py 
```

# Extract GIs from assigned OTUs (here, blast example)
```
cd ../../testrun
grep -o -e "gi|.*|gb"  out/TestData.OTU.blast.charon \                                           
        | grep -o -e "[0-9]*" \                                                                                     
        | python ../data/gi2acc/gi2accession.py -d ../data/gi2acc/gi2acc_lmdb.db \   
        | awk 'NF' \                                                                                                   
        > out/TestData.OTU.blast.charon.gi2acc.csv
```

Now we can use this file `TestData.OTU.blast.charon.gi2acc.csv` to match up our OTU GIs to their accessions

**Comment by ekrell** [comment]
oops. I spent quite a while setting up a gi 2 accession database to make the above happen, but I just noticed that our OTU files already have both GI and Accession... 

So all I need to do is re-write the R script `OTUvsTubes` to use accessions instead of GI.
This requires a library that understands accessions, which `CHNOSZ` does not.
I tried `taxonomizr`, but ran into an error. Possibly from overfilling `/tmp` ?

**Comment by ekrell** [comment]
Found the band-aid: the `taxizedb` R package.
Works much like one from `CHNOSZ`. but with easier to work with output, so I was able to simplify some loops. 

I am output `OTUvsTubes` files again. Did not even need to replace GIs with Accessions. From what I can tell, `taxizedb` understands either. 

Before I close issue:
- Update README
- Push updated OTUvsTubes R script
- Make sure it works on Chris's computer

Remaining issues:
- Still using deprecated GIs. While the pipeline functions, are we getting up-to-date data without accessions?
- The `CHNOSZ` package gave superclass, subclass, etc. I think these aren't critical, but were convenient to have in the table.
